### PR TITLE
fix(ThoughtChain): remove duplicate className and default contentOpen to false

### PR DESCRIPTION
**Pull request by feoyang**
## Summary

Two minor code-quality fixes for the ThoughtChain component:

1. **Duplicate className**: `className` was passed twice to `clsx()` in the root element's merged class computation (copy-paste error in `index.tsx`).

2. **contentOpen undefined**: `expandedKeys?.includes(key)` returns `undefined` when `expandedKeys` is not provided. Added `?? false` to ensure a proper boolean is passed to CSSMotion's `visible` prop (`Node.tsx`).

## Test plan

- [x] Added test: `className` appears only once in root element's classList
- [x] Added test: collapsible items are collapsed by default when no expandedKeys provided
- [x] All existing tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * 修复 ThoughtChain 组件中自定义 className 被重复应用的问题，确保自定义类仅附加一次于根元素。
  * 修复折叠节点默认展开行为：在未提供 expandedKeys 时，节点内容默认收起（contentOpen 为 false）。

* **Tests**
  * 新增测试：验证自定义类名不重复注入及在未传入展开键时折叠节点内容不可见。

### CODE REVIEW

File: packages/x/components/thought-chain/index.tsx [hunk 1]
```diff
@@ -59,7 +59,6 @@ const ForwardThoughtChain = React.forwardRef<any, ThoughtChainProps>((props, ref
     rootClassName,
     hashId,
     cssVarCls,
-    className,
```
Comment (kimteayon) [inline packages/x/components/thought-chain/index.tsx:62]: 这个应该不要删除吧

Comment (feoyang) [reply packages/x/components/thought-chain/index.tsx:62]: 这一行的 `className` 是和第 55 行同一个变量的重复引用 — 同一个 `className` prop 在 `clsx()` 参数中出现了两次（55 行和 62 行），`clsx` 只会把非假值拼接成字符串，传入重复的同一个变量没有任何语义作用，只会在最终 `class` 属性里产生重复的 token。

对比 antdx 其他组件，都是只传一次 `className`：
- `Bubble.tsx`（L78）
- `Sender.tsx`（L105）
- `Conversations/index.tsx`（L164）
- `Prompts/index.tsx`（L156）

ThoughtChain 是唯一的例外。通过 `git log` 看这一行是创建文件首个 commit（9c284f3）时就存在的，应该是从其他组件复制时多留的一行。

我刚刚也补充了一个更严谨的测试（@coderabbitai 提了很好的建议 — 原来的 `classList` 检查因为 DOMTokenList 自动去重而失效），现在的测试基于原始 `class` 属性字符串 + 正则，能在 bug 回归时真正 fail。

如果这行有我不知道的其它用途需要保留，请告知，我再调整。

File: packages/x/components/thought-chain/__tests__/index.test.tsx [hunk 2]
```diff
@@ -202,4 +202,14 @@ describe('ThoughtChain Component', () => {
...
   });
+
+  it('should not have duplicate className on root element', () => {
+    const { container } = render(<ThoughtChain className="my-custom-class" items={items} />);
+
+    const root = container.querySelector('.ant-thought-chain');
+    // className should appear only once in the class list
+    const classList = Array.from(root!.classList);
+    const count = classList.filter((c) => c === 'my-custom-class').length;
+    expect(count).toBe(1);
```
[the remark this answers is not in the document]
Comment (feoyang) [reply packages/x/components/thought-chain/__tests__/index.test.tsx:216]: Thanks for the catch! You're absolutely right — `DOMTokenList` de-duplicates tokens per the DOM spec, so the original assertion using `classList` couldn't actually detect the bug. 

I've pushed a fix in 3d609ff0 that reads the raw `class` attribute via `getAttribute('class')` and counts token occurrences via regex. I also verified the test correctness by reintroducing the duplicate locally — the test now fails with `Expected: 1, Received: 2`, confirming it catches the regression.

**Review by kimteayon — APPROVED** [review]
